app/models/project.rb

```diff
@@ -1,4 +1,18 @@
 class Project < ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def average_years_experience
+    total_experience = contestants.sum(:years_of_experience)
```

Comment:
Whenever you find yourself doing calculations in a model, try to pause and look if an AR method already exists to help you with this! `contestants.average(:years_of_experience)` can help you out here and I believe it's smart enough to handle edge cases in the same way your method does.